lib/utils/job-utils/command-parser.js

```diff
@@ -1246,6 +1247,74 @@ function commandParserFactory(Logger, Promise, _) {
         }));
     };
 
+    CommandParser.prototype[lsfileinfo] = function(data){
+        if(data.error){
+            return Promise.resolve({ source: 'lsfileinfo', error: data.error });
+        }
+        try {
+            var lines = data.stdout.split('\n');
+            lines.splice(0, 1);
+            var parsed = [];
+            _.forEach(lines, function(line){
+                if(!line){
+                    return;
+                }
+                var properties = line.split(/\s+/);
+                if(properties[8] === '.' || properties[8] === './' || properties[8] === '..' || properties[8] === '../'){
+                    return;
+                }
+                var item = {};
+                item.link_count = properties[1];
+                item.owner = properties[2];
+                item.group = properties[3];
+                item.size = properties[4];
+                item.modifed_time = properties[5] + ' ' + properties[6] + ' ' + properties[7];
+                item.name = properties[8];
+                var permissions = properties[0];
+                var fileType = permissions.substring(0, 1);
+                var userPermissions = permissions.substring(1, 4);
+                var groupPermissions = permissions.substring(4, 7);
+                var otherPermissions = permissions.substring(7, 10);
+                item.permission = {
+                    user: userPermissions,
+                    group: groupPermissions,
+                    other: otherPermissions
+                };
+                switch(fileType){
+                    case '-':
+                        item.file_type = 'file';
+                        break;
+                    case 'd':
+                        item.file_type = 'dir';
+                        break;
+                    case 'b':
+                        item.file_type = 'block file';
+                        break;
+                    case 'c':
+                        item.file_type = 'character device file';
+                        break;
+                    case 'p':
+                        item.file_type = 'pipe file';
+                        break;
+                    case 'l':
+                        item.file_type = 'link';
+                        var nameSplits = properties[8].split(' ');
+                        item.name = nameSplits[0];
+                        item.link_origin = nameSplits[2];
+                        break;
+                    case 's':
+                        item.file_type = 'socket file';
+                        break;
```

Comment:
great to enumerate all file types here

Reply:
suggest that encapsulate the file type judgment to a function.